.github/workflows/imports.yml

```diff
@@ -24,6 +24,25 @@ env:
 
 jobs:
 
+  no-deps-install:
+    runs-on: ubuntu-latest
+    steps:
+
+    - uses: actions/checkout@v3
+
+    - uses: actions/setup-python@v4
+      with:
+        python-version: ${{ env.PYTHON_VERSION }}
+
+    - name: Install Haystack with no dependencies
+      run: |
+        pip install git+https://github.com/ZanSara/generalimport.git@handling-classes
+        pip install pydantic tenacity pandas
```

Comment:
Why do we install these three here?

Reply:
Because they're the only ones that still absolutely need to be installed to get `import haystack` to work after you install it with `pip install --no-deps farm-haystack`. I don't think we should really make an extra for now,  because Haystack is not yet usable at this stage. It's more about making sure that all other deps are really optional and can be extracted easily once we decide whether to do it or not.

By the way I was not sure this import test should go in main. Wdty?

Reply:
So we're not actually testing without dependencies. 😅 

Why even test this? I don't really see the need to install Haystack without dependencies.

Reply:
This is a stress test for `generalimport` to make sure no Haystack dependency, however "core" it is, breaks the library or surfaces any weird corner case at import time. `generalimport` is not a very widely used library yet, so I figured an extra check could make us feel a bit more confident that we can actually make optional **every** Haystack dependency, in case we ever need that. Right now it's not granted. I found a lot of oddities in `generalimport` by trying to take out `transformers`, `torch` and `sqlalchemy`. So at least for testing it was really useful.

But I'm not sure if gives a lot of value in general. Personally I'd run it with the integration tests, for example, or once a day in `main`, but that's your call. If it seems like it brings no value let's drop it :+1:

Reply:
If it's a stress test for `generalimport` I'd remove it completely. 

If this fails and we can't fix the direct cause of the issue it's not worth in my opinion.

Reply:
> If this fails and we can't fix the direct cause of the issue it's not worth in my opinion.

Well, technically we _can_ fix the cause :smile: I've made plenty PRs to `generalimport` already, and if we're introducing a new library that breaks it, it would be nice to hold merging the PR on Haystack until we're sure that generalimport caught up or at least that a solution is possible (sometimes it isn't). 

Anyway I'll remove it for now :+1:

Reply:
In any case I'd prefer if it's something tested on `generalimport` side. Also I think it's quite silly that we test Haystack with no dependencies, it's not something we should support.